test/store.test.js

```diff
@@ -0,0 +1,239 @@
+  getSatnavBucketInfo,
+  getCarparkBucketInfo,
+  getDynamoDb,
+} from './helpers/deployment.js'
+import { createMailSlurpInbox, setupNewClient, getServiceProps } from './helpers/up-client.js'
+import { randomFile } from './helpers/random.js'
+import { getMetrics, getSpaceMetrics } from './helpers/metrics.js'
+
+test.before(t => {
+  t.context = {
+    apiEndpoint: getApiEndpoint(),
+    metricsDynamo: getDynamoDb('admin-metrics'),
+    spaceMetricsDynamo: getDynamoDb('space-metrics'),
+    rateLimitsDynamo: getDynamoDb('rate-limit')
+  }
+})
+
+// Integration test for all uploading flow with `store/add`
+test('store protocol integration flow', async t => {
+  const stage = getStage()
+  const inbox = await createMailSlurpInbox()
+  const client = await setupNewClient(t.context.apiEndpoint, { inbox })
+  const serviceProps = getServiceProps(client, t.context.apiEndpoint, StoreCapabilities.add.can)
+  const spaceDid = client.currentSpace()?.did()
+  if (!spaceDid) {
+    throw new Error('Testing space DID must be set')
+  }
+
+  // Get space metrics before store/add
+  const spaceBeforeStoreAddMetrics = await getSpaceMetrics(t, spaceDid, SPACE_METRICS_NAMES.STORE_ADD_TOTAL)
+  const spaceBeforeStoreAddSizeMetrics = await getSpaceMetrics(t, spaceDid, SPACE_METRICS_NAMES.STORE_ADD_SIZE_TOTAL)
+
+  // Get metrics before upload
+  const beforeOperationMetrics = await getMetrics(t)
+  const beforeStoreAddTotal = beforeOperationMetrics.find(row => row.name === METRICS_NAMES.STORE_ADD_TOTAL)
+  const beforeStoreAddSizeTotal = beforeOperationMetrics.find(row => row.name === METRICS_NAMES.STORE_ADD_SIZE_TOTAL)
+
+  const s3Client = getAwsBucketClient()
+  const r2Client = getCloudflareBucketClient()
+
+  const file = await randomFile(100)
+
+  // Encode file as Unixfs and perform store/add
+  const blocksReadableStream = UnixFS.createFileEncoderStream(file)
+  /** @type {import('@web3-storage/upload-client/types').CARLink[]} */
+  const shards = []
+  /** @type {import('@web3-storage/upload-client/types').AnyLink | undefined} */
+  let root
+
+  await blocksReadableStream
+    .pipeThrough(new ShardingStream())
+    .pipeThrough(
+      new TransformStream({
+        async transform(car, controller) {
+          const bytes = new Uint8Array(await car.arrayBuffer())
+          // Invoke store/add and write bytes to write target
+          const cid = await Store.add(serviceProps.conf, bytes, { connection: serviceProps.connection })
+
+          const { version, roots, size } = car
+          controller.enqueue({ version, roots, size, cid })
+
+        }
+      })
+    )
+    .pipeTo(
+      new WritableStream({
+        write(meta) {
+          root = root || meta.roots[0]
+          shards.push(meta.cid)
+        },
+      })
+    )
+
+  if (root === undefined) throw new Error('missing root CID')
+  t.is(shards.length, 1)
+
+  // Invoke upload/add
+  await Upload.add(serviceProps.conf, root, shards, { connection: serviceProps.connection })
```

Comment:
Why not use an older version of the client for this?

Reply:
I attempted to do so, but there were some dependencies collision and I opted to get this similarly to `blob` tersts